delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/Resources.scala

```diff
@@ -0,0 +1,305 @@
+      case e: ResourceDeprecated => deprecated(e)
+    }
+  }
+
+  @SuppressWarnings(Array("OptionGet"))
+  private[delta] def evaluate(
+      fetchSchema: ResourceRef => UIO[Option[SchemaResource]]
+  )(state: ResourceState, cmd: ResourceCommand)(implicit
+      clock: Clock[UIO] = IO.clock,
+      rcr: RemoteContextResolution
+  ): IO[ResourceRejection, ResourceEvent] = {
+
+    def asGraph(id: Iri, expanded: ExpandedJsonLd): IO[ResourceRejection, Graph] =
+      expanded.toGraph.leftMap(err => ResourceJsonLdPayloadRejection(Some(id), err))
+
+    def validate(schemaRef: ResourceRef, id: Iri, graph: Graph): IO[ResourceRejection, Unit] = {
+      if (schemaRef.iri == schemas.resources) IO.unit
+      else
+        for {
+          schema <- extractSchema(schemaRef)
+          report <- ShaclEngine(graph.model, schema.graph.model, validateShapes = false, reportDetails = true)
+                      .leftMap(ResourceShaclEngineRejection(id, schemaRef, _))
+          result <- if (report.isValid()) IO.unit else IO.raiseError(InvalidResource(id, schemaRef, report))
+        } yield result
+    }
+
+    def extractSchema(schemaRef: ResourceRef): IO[ResourceRejection, Schema] =
+      fetchSchema(schemaRef).flatMap {
+        case Some(schema) if schema.deprecated => IO.raiseError(SchemaIsDeprecated(schemaRef))
+        case Some(schema)                      => IO.pure(schema.value)
+        case None                              => IO.raiseError(SchemaNotFound(schemaRef))
+      }
+
+    def create(c: CreateResource) =
+      state match {
+        case Initial =>
+          for {
+            graph <- asGraph(c.id, c.expanded)
+            _     <- validate(c.schema, c.id, graph)
+            types  = c.expanded.types.toSet
+            t     <- IOUtils.instant
+          } yield ResourceCreated(c.id, c.project, c.schema, types, c.source, c.compacted, c.expanded, 1L, t, c.subject)
+
+        case _ => IO.raiseError(ResourceAlreadyExists(c.id))
+      }
+
+    def update(c: UpdateResource) =
+      state match {
+        case Initial                                         =>
+          IO.raiseError(ResourceNotFound(c.id))
+        case s: Current if s.rev != c.rev                    =>
+          IO.raiseError(IncorrectRev(c.rev, s.rev))
+        case s: Current if s.deprecated                      =>
+          IO.raiseError(ResourceIsDeprecated(c.id))
+        case s: Current if c.schemaOpt.exists(_ != s.schema) =>
+          IO.raiseError(ResourceSchemaUnexpected(s.id, c.schemaOpt.get, s.schema))
+        case s: Current                                      =>
+          for {
+            graph <- asGraph(c.id, c.expanded)
+            _     <- validate(s.schema, c.id, graph)
+            types  = c.expanded.types.toSet
+            time  <- IOUtils.instant
+          } yield ResourceUpdated(c.id, c.project, types, c.source, c.compacted, c.expanded, s.rev + 1, time, c.subject)
+
+      }
+
+    def tag(c: TagResource) =
```

Comment:
we don't verify deprecation status here; is this the same behaviour from 1.4.x?

Reply:
Yes you're right